test/data/azure/get_certs_for_model.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Curl + openssl has issues reading the system certificates on Android.
+# Pending a better solution we create an in-memory certificate store from certificates included in the model.
+#
+# The certificates must be added to the first Azure operator in the model in an attribute called 'x509_certificates'.
+# The user must determine the correct certificates for their scenario, and add them to the model.
+# The PEM file from https://curl.se/docs/caextract.html may be used.
+#
+# Include this file in the python script that is creating your model with Azure custom operators.
+# Get the value to use in the 'x509_certificates' attribute from eith a file (call get_certs_from_file) or 
+# a url (call get_certs_from_url)
+# 
+# See create_openai_whisper_transcriptions.py for example usage.
+#
+# Notes:
+#
+# - Supposedly if openssl uses md5 hashing for the certificates in /system/etc/security/cacerts it should work, but
+# a patched version of openssl with this change still failed.
+# - The 'better' solution might be to use boringssl instead of openssl as it handles the certificate format in
+# /system/etc/security/cacerts, although even that is potentially problematic as there's no versioning of boringssl.
+
+import pathlib
+import tempfile
+
+
+def _get_certs_from_input(input_data):
+    certs = None
+
+    # strip out everything except the certs as per https://curl.se/libcurl/c/cacertinmem.html example.
+    with tempfile.TemporaryFile(mode='w+', encoding='utf-8') as out:
+        in_cert = False
+        num_certs = 0
+        for line in input_data.readlines():
+            if not in_cert:
+                in_cert = "-----BEGIN CERTIFICATE-----" in line
+                if in_cert:
+                    num_certs += 1
+
+            if in_cert:
+                out.write(line)
+                in_cert = "-----END CERTIFICATE-----" not in line
+
+        assert num_certs > 0
+        assert not in_cert  # mismatched begin/end if not false
+        print(f"Processed {num_certs} certificates")
+
+        # rewind and return as UTF-8 string
+        out.seek(0)
+        certs = out.read()
```

Comment:
do we need to write a temporary file? could we write into something like [`StringIO`](https://docs.python.org/3/library/io.html#io.StringIO) instead?